framework/include/actions/ActionWarehouse.h

```diff
@@ -134,6 +135,18 @@ class ActionWarehouse : public ConsoleStreamInterface
     }
     return *p;
   }
+  template <class T>
```

Comment:
Why this function needs to be templated, wouldn't it always be returning a `PhysicsBase` or derived class?

Reply:
Thanks to the template you can get the specific type you need

This saves you a dynamic cast which is slow